app/models/term.rb

```diff
@@ -26,6 +26,26 @@ def tests
     assignments.where(assignment_type: "test")
   end
 
+  def accepted_tasks_count
+    tasks.map{ |task| task.accepted? }.count
+  end
+
+  def notes_count
+    tasks.map{ |task| task.notes_count }.sum
+  end
+
+  def submissions_count
+    tasks.map{ |task| task.submissions_count }
```

Comment:
I don't think `submissions_count` is a suitable name here. Can't see any count on results of `map`.